node/bindnode/infer.go

```diff
@@ -285,7 +285,13 @@ func inferGoType(typ schema.Type) reflect.Type {
 }
 
 // from IPLD Schema field names like "foo" to Go field names like "Foo".
-func fieldNameFromSchema(name string) string {
+func inferFieldNameFromSchema(name string) string {
+	// Hard-code a few common names to be inferred better.
+	// TODO: is there a good and small library collecting such names?
+	switch name {
+	case "id":
+		return "ID"
```

Comment:
I'm _meh_ about this one, feels like a stretch to be doing this. `Id` isn't the worst thing to end up with is it? If you go down this path you're looking at having an entry for every known acronym out there.

Reply:
+1 tbh.  I don't relish anything that gives me flashbacks to Ruby/Rail's "Inflector".

Reply:
I definitely don't want to maintain this long-term, but without it, graphsync's schema doesn't quite work. Unless we rename its Go `ID` to `Id`, which feels unnecessary.

We can definitely do better on our end, and that's https://github.com/ipld/go-ipld-prime/issues/329. I don't want to block us for another two weeks, hence the temporary internal hack.

Reply:
On second thought, it sounds like on graphsync I'll be using separate types for bindnode, so I don't need to worry about `Id` vs `ID` anymore. I'll undo this.

Reply:
the commit is now entirely removed.